runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -238,6 +239,23 @@ private static Stream<KeyedGetDataRequest> toKeyedGetDataRequestStream(
                     .build());
   }
 
+  /**
+   * Returns the current aggregate {@link GetWorkBudget} that is active on the user worker. Active
+   * means that the work is received from Windmill, being processed or queued to be processed in
+   * {@link ActiveWorkState}, and not committed back to Windmill.
+   */
+  synchronized GetWorkBudget currentActiveWorkBudget() {
```

Comment:
depending on how often this is called, it might be better to keep a running total as we add/remove things instead of calculating by iterating over everything each time this is called.

Reply:
good idea
added so that we increment when work is activated (ready to execute or queued)
and decrement when work is completed.